src/nvim/getchar.c

```diff
@@ -3435,7 +3434,8 @@ static void showmap(mapblock_T *mp, bool local)
   size_t len = 1;
 
   if (message_filtered(mp->m_keys)
-      && mp->m_str != NULL && message_filtered(mp->m_str)) {
+      && message_filtered(mp->m_str[0] != NUL ? mp->m_str : (char_u *)"<Nop>")
```

Comment:
This looks a bit inconsistent, as filtering other `<>` key notations doesn't work.

Reply:
I realized that after adding the <Nop>. I was actually thinking of making them filterable too . Should we ?

Reply:
In `map.txt`:
```
The |:filter| command can be used to select what mappings to list.  The
pattern is matched against the {lhs} and {rhs} in the raw form.
```

Reply:
I think you can limit this PR to fixing the displaying of rhs `<Nop>` mappings only, and add a test for it.

Reply:
You can't generate <Cmd> / <Lua> / <Nop> key in that manner, right ?

Reply:
For `<Nop>` one can use `:filter /^$/ map`. For `<Cmd>` one can use `:execute`. You haven't added a `<Lua>` notation yet.